lib/Inst/Inst.cpp

```diff
@@ -155,8 +155,15 @@ std::string ReplacementContext::printInst(Inst *I, llvm::raw_ostream &Out,
     case Inst::UMulO:
       break;
     default: {
-      Out << "%" << InstName << ":i" << I->Width << " = "
-          << Inst::getKindName(I->K) << OpsSS.str();
+      if (I->K == Inst::Var && (I->KnownZeros.getZExtValue() || I->KnownOnes.getZExtValue())) {
+        Out << "%" << InstName << ":i" << I->Width << " = "
+            << Inst::getKindName(I->K) <<  " "
+            << Inst::getKnownBitsPattern(I->Width, I->KnownZeros, I->KnownOnes)
```

Comment:
you can avoid some needless code duplication here if you conditionally print this string instead of conditionally printing the entire line